samples/go/photo-index/main.go

```diff
@@ -55,38 +56,59 @@ func index() (win bool) {
 
 	inputs := []types.Value{}
 	for i := 0; i < flag.NArg(); i++ {
-		if dataset.DatasetFullRe.MatchString(flag.Arg(i)) {
-			ds := dataset.NewDataset(db, flag.Arg(i))
-			v, ok := ds.MaybeHeadValue()
-			if ok {
-				inputs = append(inputs, v)
-				continue
-			}
+		p, err := spec.NewAbsolutePath(flag.Arg(i))
```

Comment:
I think this could be useful as a method in the specs package. It seems like there might be lots of programs that want to take a number of paths as arguments. Something like: 
```ResolvePaths(db datas.Database, paths []string) (ValueSlice, err)```